controllers/reportController.js

```diff
@@ -0,0 +1,103 @@
+const {
+    createReport,
+    getAllReports,
+    getReportById
+  } = require('../lib/modelManagers/reportModel'),
+  {
+    failureResponse,
+    successResponse
+  } = require('../lib/utils/helper_function'),
+  {
+    OK_CODE,
+    RESOURCE_CREATED_CODE,
+    SERVER_ERROR_CODE,
+    NO_REPORT_FOUND,
+    RETRIEVED_REPORT,
+    REPORT_CREATED,
+    SERVER_ERROR_MESSAGE
+  } = require('../constants/index'),
+  pagination = require('../lib/utils/pagination/paginationHelper');
+/**
+ * @description - handles report
+ *
+ * @class ReportController
+ */
+class ReportController {
+  /**
+   * @description - handle create report
+   *
+   * @static
+   * @param {any} req Request Object
+   * @param {any} res Response Object
+   * @returns {object} - Returns created report
+   * @memberOf ReportController
+   */
+  static async createAReport(req, res) {
+    try {
+      const { articleId } = req.params;
+      const userId = req.decodedToken.payLoad;
+      const {
+        reportTagId, message
+      } = req.body;
+
+      const reportData = {
+        userId,
+        articleId,
+        reportTagId,
+        message
+      };
+      const createdReport = await createReport(reportData);
+      if (createdReport) {
+        return successResponse(res, REPORT_CREATED, createdReport, RESOURCE_CREATED_CODE);
+      }
+    } catch (error) {
+      return failureResponse(res, SERVER_ERROR_MESSAGE, SERVER_ERROR_CODE);
+    }
+  }
+
+  /**
+ * @description - handles get all tags
+ *
+ * @static
+ * @param {any} req Request Object
+ * @param {any} res Response Object
+ * @returns {Object} returns object of reports
+ * @memberOf ReportController
+ */
+  static async getReports(req, res) {
+    const { limit, offset } = pagination(req.query);
+    try {
+      const reports = await getAllReports(limit, offset);
+      if (reports.length > 0) {
+        return successResponse(res, RETRIEVED_REPORT, reports, OK_CODE);
+      }
```

Comment:
```
 if (reports.length > 0) {
        return successResponse(res, RETRIEVED_REPORT, reports, OK_CODE);
      }
```
Any specific reason for the above?

Reply:
To check the length of the data in the database and if it is zero (0) to return a 200 status code but a message indicating success but no record found in the database.